🌐✨i18n for newsletter strings

cathysarisky opened the pull request:
no issue

This PR adds the ability to translate the strings that appear in the newsletter as boilerplate text, using i18next.

Variables are in single mustaches ( `{date}` ) in the translation strings (rather than `{{date}}`), because these strings occur both the email template.hbs and also .js files.  That necessitated a separate namespace.

This PR also includes changes to the newsletter button ("more like this", "less like this", "comment") that were previously delivered on desktop as images that included the text.  @sanne-san provided a rework that removed text-as-image from the desktop buttons, and allows more shared code between the two layouts, along with making the buttons translatable.

Example usage - handlebars
```
<h3 class="latest-posts-header">{{t 'Keep reading'}}</h3>

{{{t 'By {authors}' authors=post.authors }}} 
```
(NOTE: triple { required because of possible & )

Example usage - javascript
```
                getValue: (member) => {
                    if (member.status === 'comped') {
                        return t('complimentary');
                    }
                    if (this.isMemberTrialing(member)) {
                        return t('trialing');
                    }
                    // other possible statuses: t('free'), t('paid') //
                    return t(member.status);
                }
```

cathysarisky committed:
initial commit iwth date error

cathysarisky committed:
almost there!

cathysarisky committed:
fixed!

cathysarisky committed:
Merge branch 'main' into newsletter-t

cathysarisky committed:
ah lint

cathysarisky committed:
Merge branch 'newsletter-t' of https://github.com/cathysarisky/Ghost into newsletter-t

cathysarisky committed:
dangit lint

cathysarisky committed:
stray space = failed test :(

cathysarisky committed:
add a test

cathysarisky committed:
Merge branch 'main' into newsletter-t

cathysarisky committed:
date i18n

cathysarisky committed:
Merge branch 'newsletter-t' of https://github.com/cathysarisky/Ghost into newsletter-t

cathysarisky committed:
belly button lint?

cathysarisky committed:
issues with imports?

cathysarisky committed:
import issues

cathysarisky committed:
build error?

cathysarisky committed:
tweak imports again?

cathysarisky committed:
hating importing.

vikaspotluri123 commented on ghost/email-service/i18n-setup.js:
From what I've seen in the past, you want to limit the files in the package root - it might make sense to move this to the lib folder

vikaspotluri123 commented on ghost/email-service/i18n-setup.js:
```diff
@@ -0,0 +1,15 @@
+const i18nLib = require('../i18n');
```
I _think_ you don't want to import files across package boundaries - you would need to add the package as a dependency and import from it

vikaspotluri123 commented on ghost/email-service/i18n-setup.js:
```diff
@@ -0,0 +1,15 @@
+const i18nLib = require('../i18n');
+const settingsCache = require('../core/core/shared/settings-cache');
```
You might need to make a part of the email renderer since I think the `settingsCache` needs to be provided as a dependency rather than imported

vikaspotluri123 commented:
Definitely take what I said with a grain of salt! Someone should confirm my observations 🙈

cathysarisky committed:
try building, probably fail lint.

cathysarisky committed:
only one failing test?

cathysarisky committed:
t fun.

cathysarisky committed:
e2e test fix?

cathysarisky committed:
test fussing

cathysarisky committed:
test wrangling

cathysarisky committed:
better test

cathysarisky committed:
lint is hard

cathysarisky committed:
make buttons translatable

cathysarisky committed:
preliminary update of snapshots for translatable buttons

cathysarisky commented on package.json:
```diff
@@ -128,5 +128,6 @@
     "typescript": "5.4.5"
-  }
+  },
+  "packageManager": "yarn@1.22.21+sha1.1959a18351b811cdeedbd484a8f86c3cc3bbaf72"
```
```suggestion
```

cathysarisky commented on package.json:
```diff
@@ -128,5 +128,6 @@
     "ts-node": "10.9.2",
     "typescript": "5.4.5"
-  }
+  },
```
```suggestion
  }
```

cathysarisky committed:
argh, package.json...

cathysarisky commented:
@sanne-san , if you have time to look at this, I'd appreciate it! :)

cathysarisky committed:
fix dates... again.

cathysarisky committed:
Merge branch 'newsletter-t' of https://github.com/cathysarisky/Ghost into newsletter-t

cathysarisky committed:
added a creation date - my mock is too good, and needs one to make the snapshots work!

cathysarisky committed:
date localization problem fixed?

cathysarisky committed:
and the snap

cathysarisky committed:
test tweaks

sanne-san committed:
Updated feedback button styling

No issue
- Removed all mobile feedback button code
- Updated styling of feedback buttons to display in a single row on desktop and column on mobile

9larsons committed:
Cover last lines in unit test

cathysarisky committed:
fix broken test

cathysarisky committed:
more broken tests

cathysarisky committed:
fix snapshots in places I didn't know we had snapshots.

cathysarisky committed:
more snapshots.

cathysarisky committed:
Merge branch 'main' into newsletter-t

cathysarisky committed:
remove unused tpl, improve handling of dynamic strings (sort of)

cathysarisky committed:
must... write... better... code

cathysarisky committed:
Merge branch 'main' into newsletter-t

cathysarisky committed:
i18nmerge fussing

cathysarisky committed:
Merge branch 'main' into newsletter-t

cathysarisky committed:
trying to make i18n parser play nice

cathysarisky committed:
Merge branch 'newsletter-t' of https://github.com/cathysarisky/Ghost into newsletter-t
